ambry-tools/src/main/java/com.github.ambry/store/DumpIndexTool.java

```diff
@@ -506,8 +705,22 @@ public IndexProcessingResults processIndex(File replicaDir, Set<StoreKey> filter
           }
         }
       }
+      Set<StoreKey> duplicateKeys = new HashSet<>();
+      if (detectDuplicatesAcrossKeys) {
+        logger.info("Converting {} store keys...", keyToState.size());
+        Map<StoreKey, StoreKey> conversions = storeKeyConverter.convert(keyToState.keySet());
+        logger.info("Store keys converted!");
+        conversions.forEach((k, v) -> {
+          if (!k.equals(v) && keyToState.containsKey(v)) {
```

Comment:
Correct me if I misunderstood. If we have A1 and A2 on a single store. After conversion, the results are <A1, A2> and <A2, A2>. So this piece of code is gonna add A2 into duplicateKeys? Am I right?